src/uct/cuda/base/cuda_md.c

```diff
@@ -145,14 +152,30 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_base_mem_query,
 
         if (mem_attr->field_mask & (UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH |
                                     UCT_MD_MEM_ATTR_FIELD_BASE_ADDRESS)) {
-            cu_err = cuMemGetAddressRange((CUdeviceptr*)&base_address,
-                                          &alloc_length, (CUdeviceptr)address);
+            cu_err = cuDeviceGetName(dev_name, UCT_CUDA_DEV_NAME_MAX_LEN,
```

Comment:
1. use sizeof(dev_name) instead of UCT_CUDA_DEV_NAME_MAX_LEN
2. can we somehow check the device by the memory buffer pointer (in case current cuda context changed)?

Reply:
@yosefe For 2., we do get the device ordinal from the memory buffer pointer (you'll see that cuda_device is one of the attributes we query using the pointer). Later we use the same cuda_device to get the device name. Did I misunderstand the question?